statbot/sql.py

```diff
@@ -915,56 +911,53 @@ def upsert_member(self, trans, member):
         self._delete_role_membership(trans, member)
         self._insert_role_membership(trans, member)
 
-    # Emojis (TODO)
+    # Emojis
     def add_emoji(self, trans, emoji):
-        # pylint: disable=unreachable
-        raise NotImplementedError
-
-        values = emoji_values(emoji)
-        id = values['emoji_id']
-        if id in self.emoji_cache:
-            self.logger.debug(f"Emoji {id} already inserted.")
+        data = EmojiData(emoji)
+        if data.cache_id in self.emoji_cache:
+            self.logger.debug(f"Emoji {data} already inserted.")
```

Comment:
can we talk about how this plus `_Transaction` just re-invents sqlalchemy sessions?

Reply:
Yes, but according to the SQLAlchemy docs, sessions should be a more permanent thing, removed from individual clusters of operations. Given that we need this handle to be created and closed for every single operation statbot does, using transactions is more suitable.